### PR TITLE
Implement xpns_log()

### PR DESCRIPTION
**Pull request by ReDTerraN**
Introducing xpns_log()
This will replace following functions
- xpns_msg()
- xpns_msg_info()
- xpns_msg_warning() 
- xpns_msg_error()
- xpns_msg_debug() 

This will consolidate our code and make it easier as its all in one function. 

**Features**
 loglevels is now called via an positional argument, eg:
`xpns_log "error" "${_dir} is not writable."`

we now have following values we can call:

-  info
-  warning
-  error
-  debug

**New look to the logoutput**
It now comes with a new updated output to improve readability

old vs new
`xpanes:Info: This is an info  message`
`xpanes:Debug: [2023-04-18_15:24:33]:main:debug example`

`[xpanes:info] This is an info message `
`[xpanes:debug:2023-04-18_15:24:33:main] debug example`
 
 **Ability to parse multiple inputs**
 
 old function had the limitations to only parse the first positional argument it acquired. With xpns_log we can make how many arguments we wish.
 
 For example:
 
old and new
`xpns_msg_info "this is an info message" "and here is more information"`
` xpns_log "info" "This is an info message" "and here is more information"`

this provides following input:

old
`xpanes:Info: this is an info  message`
new
`[xpanes:info] This is an info message and here is more information`

I have tested the function on all the input in xpanes, but should we ever get an issue with parsing the loglevel, we have a following output:
` [xpanes:internal error] invalid log type, if you get this error. Please file an issue on github: https://github.com/greymd/tmux-xpanes/issues `

Referring to setup an issue on the project if it ever occurs. 
I have also made unit testing for the function to make sure we catch any issues.

Cheers

### CODE REVIEW

**Comment by ReDTerraN** [comment]
hmm, my unit test and shfmt tests works locally, But it does not work in the CI/CD pipeline. I will try later this week troubleshoot why it is the case.

**Comment by ReDTerraN** [comment]
All dependencies should now have been secured. 
But im unable to satistfy shfmt and I do not get an hint to what I need to fix.

Running the command locally provides me no hint to what is needed
```
shfmt -i 2 -ci -sr -kp -d ./bin/xpanes ./*.sh
 echo $?
0

```

@greymd If you can assist that would be greatly helpful, other than that I now believe its ready to reviewed.

Thanks :)

**Comment by greymd** [comment]
> Running the command locally provides me no hint to what is needed

Oh really, ok I am taking a look..

File: bin/xpanes [hunk 1]
```diff
@@ -1856,9 +1850,9 @@ xpns_opt_checker() {
     fi
   fi
   if [[ -n "$_arg" ]] && [[ "$_arg" =~ $_pattern ]]; then
-    return 0
+                                                             return 0
```
Comment (greymd) [inline bin/xpanes]: It seems there are too many spaces.
@ReDTerraN could you import it and check how it goes?

```suggestion
    return 0
```

FYI: Here is how I detected.

```bash
$ docker run -it --rm -v $PWD:/work mvdan/shfmt:v3.1.0-alpine shfmt -i 2 -ci -sr -kp -w /work/bin/xpanes
$ git diff
diff --git a/bin/xpanes b/bin/xpanes
index fa1d2d9..2611a1f 100755
--- a/bin/xpanes
+++ b/bin/xpanes
@@ -1850,7 +1850,7 @@ xpns_opt_checker() {
     fi
   fi
   if [[ -n "$_arg" ]] && [[ "$_arg" =~ $_pattern ]]; then
-                                                             return 0
+    return                                                          0
   fi
   xpns_log "error" "Invalid argument '$_arg' for $_option option"
   exit ${XP_EINVAL}
```

**Comment by ReDTerraN** [comment]
Thanks @greymd, I have now applied the docker container to my workflow instead of running it locally. All checks should now be done.

**Pull request merged**